src/views/scans/__tests__/showAggregateReportModal.test.tsx

```diff
@@ -90,4 +90,9 @@ describe('formatSortFilterReportStats', () => {
   it('should handle a missing report', () => {
     expect(formatSortFilterReportStats()).toMatchSnapshot('missing');
   });
+
+  it('should handle misuse', () => {
+    // @ts-expect-error Incorrect type, misuse or unknown passed to function
+    expect(formatSortFilterReportStats({ results: null, diagnostics: null })).toMatchSnapshot('misuse');
```

Comment:
Is the goal here to make sure the function can handle bad input like null values, or should this kind of check be handled before calling it?

Reply:
yes, technically you could do handle this inside your API hook or the calling component...

> we reasoned not doing this because...
> -  we attempted to match existing API hooks where data is minimally transformed... generally in an attempt to broadcast a notification
> - separating it out of the component to help testing and creating a component that "just handles display". similar to dependency injection

The final reasoning went like...

In the end we saw 2 paths `before calling it`... the context of calling the function 
- in regards to the API
- in regards to a standalone unit that does something

Both paths end up relying on the function, ergo the function has the check